### PR TITLE
Fix waste render on connectivity status

### LINKED ISSUE #2418: Connectivity status should not waste-render every 400ms
## Version

`master`

## Describe the bug

When putting a `console.log()` inside the render function of `DebouncedConnectivityStatus`, it was called every 400 ms due to wasted render.

## To Reproduce

As described in the above section.

## Expected behavior

It should only render when changed.

## Additional context

This is caused by recent React functional component push.

[Bug]

### PR DESCRIPTION
**Pull request by compulim**
Fixes #2418.

### Fixed

-  Fix [#2418](https://github.com/microsoft/BotFramework-WebChat/issues/2418). Connectivity status should not waste-render every 400 ms, by [@compulim](https://github.com/compulim) in PR [#2419](https://github.com/microsoft/BotFramework-WebChat/pull/2419)

## Description

`DebouncedConnectivityStatus` is causing wasted render for every 400 ms.

## Specific Changes

This is due to misunderstanding of `useState` for assigning a function.

Instead of,

```js
const [predicate, setPredicate] = useState();

setValue(x => x === 'Hello, World!');
```

It should be,

```js
const [predicate, setPredicate] = useState();

setValue(() => x => x === 'Hello, World!');
```

This is because, if function is passed to `setXXX`, React will call the function with the existing value, and use the return value as the new state. If non-function is passed, React will directly set it as new state. This behavior is currently not documented.

---

-  [x] ~Testing Added~
   - We currently do not have a way to detect how many render done per seconds, etc. We should setup this tool.

### CODE REVIEW

File: packages/component/src/SendBox/ConnectivityStatus.js [hunk 1]
```diff
@@ -10,18 +10,30 @@ import WarningNotificationIcon from '../Attachment/Assets/WarningNotificationIco
...
+  if (interval > 0) {
+    return setTimeout(fn, interval);
+  }
+
+  fn();
+}
+
 const DebouncedConnectivityStatus = ({ interval, children: propsChildren }) => {
-  const [children, setChildren] = useState(propsChildren);
+  const [children, setChildren] = useState(() => propsChildren);
```
Comment (tdurnford) [inline packages/component/src/SendBox/ConnectivityStatus.js]: Can you explain the difference between `useState(() => propsChildren)` and `useState(propsChildren)`? I don't understand the difference.

Comment (corinagum) [reply packages/component/src/SendBox/ConnectivityStatus.js]: (I believe) the return prevents it from rendering every time - just when there's a change in value.

Comment (corinagum) [reply packages/component/src/SendBox/ConnectivityStatus.js]: because I definitely HAVEN'T made this mistake 5 billion times

Comment (compulim) [reply packages/component/src/SendBox/ConnectivityStatus.js]: Explained offline. 😉

**Pull request merged**